src/Core/GenericEqualityComparer.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace WebLinq
+{
+    public class GenericEqualityComparer<T, U> : IEqualityComparer<U>
+    {
+        private Func<U, T> contraMapper;
+        private IEqualityComparer<T> comparer;
+
+        public GenericEqualityComparer(Func<U, T> contraMapper, IEqualityComparer<T> comparer)
```

Comment:
If you reverse the argument, you could turn this entirely into an extension method and move the class as a private implementation detail of the _extending class_.